packages/sanity/src/core/versions/components/GlobalBundleMenu.tsx

```diff
@@ -0,0 +1,136 @@
+import {AddIcon, CheckmarkIcon} from '@sanity/icons'
+import {Box, Button, Flex, Menu, MenuButton, MenuDivider, MenuItem, Text} from '@sanity/ui'
+import {useCallback, useContext, useState} from 'react'
+import {useRouter} from 'sanity/router'
+
+import {
+  VersionContext,
+  type VersionContextValue,
+} from '../../../_singletons/core/form/VersionContext'
+import {type Bundle, type Version} from '../types'
+import {BUNDLES, LATEST} from '../util/const'
+import {isDraftOrPublished} from '../util/dummyGetters'
+import {CreateBundleDialog} from './dialog/CreateBundleDialog'
+import {VersionBadge} from './VersionBadge'
+
+export function GlobalBundleMenu(): JSX.Element {
+  const router = useRouter()
+
+  // eslint-disable-next-line no-warning-comments
+  // FIXME REPLACE WHEN WE HAVE REAL DATA
+  const bundles = BUNDLES
+
+  // eslint-disable-next-line no-warning-comments
+  // FIXME REPLACE WHEN WE HAVE REAL DATA
+  const {currentVersion, setCurrentVersion, isDraft} =
+    useContext<VersionContextValue>(VersionContext)
+  const [createBundleDialogOpen, setCreateBundleDialogOpen] = useState(false)
+
+  const handleBundleChange = useCallback(
+    (bundle: Version) => () => {
+      const {name} = bundle
+
+      if (isDraftOrPublished(name)) {
+        router.navigateStickyParam('perspective', '')
+      } else {
+        router.navigateStickyParam('perspective', `bundle.${name}`)
+      }
+
+      setCurrentVersion(bundle)
+    },
+    [router, setCurrentVersion],
+  )
+
+  /* create new bundle */
+
+  const handleCreateBundleClick = useCallback(() => {
+    setCreateBundleDialogOpen(true)
+  }, [])
+
+  const handleCancel = useCallback(() => {
+    setCreateBundleDialogOpen(false)
+  }, [])
+
+  const handleSubmit = useCallback(
+    () => (value: Bundle) => {
+      // eslint-disable-next-line no-console
+      console.log('create new bundle', value.name)
+
+      setCreateBundleDialogOpen(false)
+    },
+    [],
+  )
+
+  return (
+    <>
+      <MenuButton
+        button={
+          <Button mode="bleed" padding={0} radius="full">
+            <VersionBadge
+              tone={currentVersion?.tone}
+              icon={isDraft ? undefined : currentVersion?.icon}
+              openButton
+              padding={2}
+              title={isDraft ? LATEST.title : currentVersion?.title}
+            />
+          </Button>
+        }
+        id="global-version-menu"
+        menu={
+          <Menu>
+            <MenuItem
+              iconRight={isDraft ? <CheckmarkIcon /> : undefined}
+              onClick={handleBundleChange(LATEST)}
```

Comment:
Sorry, I may be misreading the code, but should this be wrapped in a function?
```suggestion
              onClick={() => handleBundleChange(LATEST)}
```

Reply:
I _think_ it shouldn't be needed since `handleBundleChange` is wrapped in a `useCallback`

Reply:
This is my mistake, I didn't see that `handleBundleChange` already returns a function. Sorry 🤦‍♀️.

Reply:
No need to apologise these things happen (especially to me, really often 🫠 )